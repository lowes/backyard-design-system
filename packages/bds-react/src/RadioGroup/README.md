# @lowes-tech/bds-react/RadioGroup

## Interfaces

- [RadioGroupProps](interfaces/RadioGroupProps.md)
- [RadioGroupValue](interfaces/RadioGroupValue.md)

## References

### default

Renames and re-exports [RadioGroup](README.md#radiogroup)

## Type aliases

### RadioGroupRef

Ƭ **RadioGroupRef**: `FormGroupRef`

## Variables

### RadioGroup

• **RadioGroup**: `BDSForwardRef`<[`RadioGroupProps`](interfaces/RadioGroupProps.md)\>

Backyard React Radio Group

`RadioGroup` allows for grouping radio inputs and syncing them together.
It extends from `FormGroup` to provide extra functionality for `Radio` components.

For `Checkbox` or `Switch` it is recommended to use `FormGroup`.

This component is useful when combined with `FormControl` to
organize the visual structure of the form input. Example:

```
 <FormControl>
     <FormHeading>Heading</FormHeading>
     <RadioGroup
         name="radios"
         defaultValue="2"
         direction="row"
     >
         <FormControlLabel
             control={<Radio value="1" />}
             label="Label 1"
         />
         <FormControlLabel
             control={<Radio value="2" />}
             label="Label 2"
         />
         <FormControlLabel
             control={<Radio value="3" />}
             label="Label 3"
         />
     </FormGroup>
     <FormHelperText>Helper Text</FormHelperText>
 </FormControl>
```

___

### RadioGroupContext

• **RadioGroupContext**: `Context`<[`RadioGroupValue`](interfaces/RadioGroupValue.md)\>

## Functions

### useRadioGroup

▸ `Const` **useRadioGroup**(): [`RadioGroupValue`](interfaces/RadioGroupValue.md)

#### Returns

[`RadioGroupValue`](interfaces/RadioGroupValue.md)
