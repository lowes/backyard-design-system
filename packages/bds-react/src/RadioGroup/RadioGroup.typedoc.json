{
	"id": 0,
	"name": "@lowes-tech/bds-react/RadioGroup",
	"kind": 1,
	"kindString": "Project",
	"flags": {},
	"children": [
		{
			"id": 26,
			"name": "default",
			"kind": 16777216,
			"kindString": "Reference",
			"flags": {},
			"target": 1
		},
		{
			"id": 2,
			"name": "RadioGroupProps",
			"kind": 256,
			"kindString": "Interface",
			"flags": {},
			"children": [
				{
					"id": 11,
					"name": "as",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"type": {
						"type": "reference",
						"typeArguments": [
							{
								"type": "intrinsic",
								"name": "any"
							}
						],
						"name": "ElementType"
					},
					"inheritedFrom": {
						"type": "reference",
						"name": "Omit.as"
					}
				},
				{
					"id": 3,
					"name": "children",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"comment": {
						"shortText": "Children of `RadioGroup`, which have access to `RadioGroupContext`\nMust contain at least one `Radio` child\nDoes not require `Radio` to be direct children"
					},
					"type": {
						"type": "reference",
						"name": "ReactNode"
					},
					"overwrites": {
						"type": "reference",
						"name": "Omit.children"
					}
				},
				{
					"id": 12,
					"name": "className",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"comment": {
						"shortText": "DOM class name"
					},
					"type": {
						"type": "intrinsic",
						"name": "string"
					},
					"inheritedFrom": {
						"type": "reference",
						"name": "Omit.className"
					}
				},
				{
					"id": 4,
					"name": "defaultValue",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"comment": {
						"shortText": "Default Value of `RadioGroup`"
					},
					"type": {
						"type": "intrinsic",
						"name": "string"
					},
					"overwrites": {
						"type": "reference",
						"name": "Omit.defaultValue"
					}
				},
				{
					"id": 13,
					"name": "direction",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"comment": {
						"shortText": "Direction to render children"
					},
					"type": {
						"type": "union",
						"types": [
							{
								"type": "literal",
								"value": "column"
							},
							{
								"type": "literal",
								"value": "row"
							}
						]
					},
					"inheritedFrom": {
						"type": "reference",
						"name": "Omit.direction"
					}
				},
				{
					"id": 5,
					"name": "name",
					"kind": 1024,
					"kindString": "Property",
					"flags": {},
					"comment": {
						"shortText": "Required\nDOM field `name` for all `Radio`s in child tree of `RadioGroup`"
					},
					"type": {
						"type": "intrinsic",
						"name": "string"
					},
					"overwrites": {
						"type": "reference",
						"name": "Omit.name"
					}
				},
				{
					"id": 14,
					"name": "spacing",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"comment": {
						"shortText": "Spacing token between form inputs"
					},
					"type": {
						"type": "intrinsic",
						"name": "string"
					},
					"inheritedFrom": {
						"type": "reference",
						"name": "Omit.spacing"
					}
				},
				{
					"id": 10,
					"name": "value",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"comment": {
						"shortText": "Value of `RadioGroupContext`\nSet by current active `Radio` child of `RadioGroup`"
					},
					"type": {
						"type": "intrinsic",
						"name": "string"
					},
					"overwrites": {
						"type": "reference",
						"name": "Omit.value"
					}
				},
				{
					"id": 6,
					"name": "onChange",
					"kind": 2048,
					"kindString": "Method",
					"flags": {
						"isOptional": true
					},
					"signatures": [
						{
							"id": 7,
							"name": "onChange",
							"kind": 4096,
							"kindString": "Call signature",
							"flags": {},
							"comment": {
								"shortText": "`onChange` function to trigger when `value` of `RadioGroupContext` changes",
								"tags": [
									{
										"tag": "argument",
										"text": "{Event} event - DOM Event of `onChange`"
									},
									{
										"tag": "argument",
										"text": "{boolean} value -  Value from DOM Event target\n"
									}
								]
							},
							"parameters": [
								{
									"id": 8,
									"name": "event",
									"kind": 32768,
									"kindString": "Parameter",
									"flags": {},
									"type": {
										"type": "reference",
										"typeArguments": [
											{
												"type": "reference",
												"name": "Element"
											}
										],
										"name": "ChangeEvent"
									}
								},
								{
									"id": 9,
									"name": "value",
									"kind": 32768,
									"kindString": "Parameter",
									"flags": {},
									"type": {
										"type": "intrinsic",
										"name": "string"
									}
								}
							],
							"type": {
								"type": "intrinsic",
								"name": "void"
							}
						}
					]
				}
			],
			"groups": [
				{
					"title": "Properties",
					"kind": 1024,
					"children": [
						11,
						3,
						12,
						4,
						13,
						5,
						14,
						10
					]
				},
				{
					"title": "Methods",
					"kind": 2048,
					"children": [
						6
					]
				}
			],
			"extendedTypes": [
				{
					"type": "reference",
					"typeArguments": [
						{
							"type": "reference",
							"name": "FormGroupProps"
						},
						{
							"type": "reference",
							"name": "RadioGroupOverrideProps"
						}
					],
					"name": "Omit"
				}
			]
		},
		{
			"id": 17,
			"name": "RadioGroupValue",
			"kind": 256,
			"kindString": "Interface",
			"flags": {},
			"comment": {
				"shortText": "Value of the `RadioGroupContext` provider"
			},
			"children": [
				{
					"id": 23,
					"name": "defaultValue",
					"kind": 1024,
					"kindString": "Property",
					"flags": {},
					"comment": {
						"shortText": "Default value of the radio group"
					},
					"type": {
						"type": "intrinsic",
						"name": "string"
					}
				},
				{
					"id": 18,
					"name": "name",
					"kind": 1024,
					"kindString": "Property",
					"flags": {},
					"comment": {
						"shortText": "Name of the radio group"
					},
					"type": {
						"type": "intrinsic",
						"name": "string"
					}
				},
				{
					"id": 22,
					"name": "value",
					"kind": 1024,
					"kindString": "Property",
					"flags": {},
					"comment": {
						"shortText": "Value of the radio group"
					},
					"type": {
						"type": "intrinsic",
						"name": "string"
					}
				},
				{
					"id": 19,
					"name": "onChange",
					"kind": 2048,
					"kindString": "Method",
					"flags": {},
					"signatures": [
						{
							"id": 20,
							"name": "onChange",
							"kind": 4096,
							"kindString": "Call signature",
							"flags": {},
							"comment": {
								"shortText": "`onChange` trigger event when `RadioGroup` changes value"
							},
							"parameters": [
								{
									"id": 21,
									"name": "event",
									"kind": 32768,
									"kindString": "Parameter",
									"flags": {},
									"type": {
										"type": "reference",
										"typeArguments": [
											{
												"type": "reference",
												"name": "Element"
											}
										],
										"name": "ChangeEvent"
									}
								}
							],
							"type": {
								"type": "intrinsic",
								"name": "void"
							}
						}
					]
				}
			],
			"groups": [
				{
					"title": "Properties",
					"kind": 1024,
					"children": [
						23,
						18,
						22
					]
				},
				{
					"title": "Methods",
					"kind": 2048,
					"children": [
						19
					]
				}
			]
		},
		{
			"id": 15,
			"name": "RadioGroupRef",
			"kind": 4194304,
			"kindString": "Type alias",
			"flags": {},
			"type": {
				"type": "reference",
				"name": "FormGroupRef"
			}
		},
		{
			"id": 1,
			"name": "RadioGroup",
			"kind": 32,
			"kindString": "Variable",
			"flags": {},
			"comment": {
				"shortText": "Backyard React Radio Group",
				"text": "`RadioGroup` allows for grouping radio inputs and syncing them together.\nIt extends from `FormGroup` to provide extra functionality for `Radio` components.\n\nFor `Checkbox` or `Switch` it is recommended to use `FormGroup`.\n\nThis component is useful when combined with `FormControl` to\norganize the visual structure of the form input. Example:\n\n```\n <FormControl>\n     <FormHeading>Heading</FormHeading>\n     <RadioGroup\n         name=\"radios\"\n         defaultValue=\"2\"\n         direction=\"row\"\n     >\n         <FormControlLabel\n             control={<Radio value=\"1\" />}\n             label=\"Label 1\"\n         />\n         <FormControlLabel\n             control={<Radio value=\"2\" />}\n             label=\"Label 2\"\n         />\n         <FormControlLabel\n             control={<Radio value=\"3\" />}\n             label=\"Label 3\"\n         />\n     </FormGroup>\n     <FormHelperText>Helper Text</FormHelperText>\n </FormControl>\n```\n"
			},
			"type": {
				"type": "reference",
				"typeArguments": [
					{
						"type": "reference",
						"id": 2,
						"name": "RadioGroupProps"
					}
				],
				"name": "BDSForwardRef"
			},
			"defaultValue": "..."
		},
		{
			"id": 16,
			"name": "RadioGroupContext",
			"kind": 32,
			"kindString": "Variable",
			"flags": {},
			"type": {
				"type": "reference",
				"typeArguments": [
					{
						"type": "reference",
						"id": 17,
						"name": "RadioGroupValue"
					}
				],
				"name": "Context"
			},
			"defaultValue": "..."
		},
		{
			"id": 24,
			"name": "useRadioGroup",
			"kind": 64,
			"kindString": "Function",
			"flags": {
				"isConst": true
			},
			"signatures": [
				{
					"id": 25,
					"name": "useRadioGroup",
					"kind": 4096,
					"kindString": "Call signature",
					"flags": {},
					"type": {
						"type": "reference",
						"id": 17,
						"name": "RadioGroupValue"
					}
				}
			]
		}
	],
	"groups": [
		{
			"title": "References",
			"kind": 16777216,
			"children": [
				26
			]
		},
		{
			"title": "Interfaces",
			"kind": 256,
			"children": [
				2,
				17
			]
		},
		{
			"title": "Type aliases",
			"kind": 4194304,
			"children": [
				15
			]
		},
		{
			"title": "Variables",
			"kind": 32,
			"children": [
				1,
				16
			]
		},
		{
			"title": "Functions",
			"kind": 64,
			"children": [
				24
			]
		}
	]
}