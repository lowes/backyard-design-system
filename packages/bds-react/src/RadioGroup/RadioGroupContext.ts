import * as React from 'react'

const RadioGroupContext = React.createContext<RadioGroupValue>(null)

/**
 * Value of the `RadioGroupContext` provider
 */
interface RadioGroupValue {
    /**
     * Name of the radio group
     */
    name: string
    /**
     * `onChange` trigger event when `RadioGroup` changes value
     */
    onChange: (event: React.ChangeEvent) => void
    /**
     * Value of the radio group
     */
    value: string
    /**
     * Default value of the radio group
     */
    defaultValue: string
}

export {
    RadioGroupContext
}

export type {
    RadioGroupValue
}

export default RadioGroupContext
