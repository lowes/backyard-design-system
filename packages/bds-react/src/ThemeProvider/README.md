# @lowes-tech/bds-react/ThemeProvider

## Namespaces

- [ThemeProvider](modules/ThemeProvider.md)

## Interfaces

- [BackyardTheme](interfaces/BackyardTheme.md)
- [ThemeProviderProps](interfaces/ThemeProviderProps.md)

## References

### default

Renames and re-exports [ThemeProvider](README.md#themeprovider)

## Type aliases

### BaseTheme

Ƭ **BaseTheme**: typeof `Theme`

## Variables

### ThemeConsumer

• **ThemeConsumer**: `React.Consumer`<[`BackyardTheme`](interfaces/BackyardTheme.md)\> = `StyledComponentsThemeConsumer`

ThemeConsumer same as 'styled-components'

___

### ThemeContext

• **ThemeContext**: `React.Context`<[`BackyardTheme`](interfaces/BackyardTheme.md)\> = `StyledComponentsThemeContext`

ThemeContext same as 'styled-components'

___

### ThemeProvider

• **ThemeProvider**: `BDSFunctional`<[`ThemeProviderProps`](interfaces/ThemeProviderProps.md)\>

Backyard React Theme Provider

 <ThemeProvider theme="light" font="fellix" someOtherContext="value">
     <Grid.Container>
         ...
             <Button />
         ...
     </Grid.Container>
 </ThemeProvider>

Custom Theme Provider wrapper to provide extra context and overrides for Backyard Themes.

## Functions

### dropShadow

▸ `Const` **dropShadow**(`shadow`, `defaultShadow?`): `string`

Converts box-shadow to drop-shadow, if necessary.

Use for `filter` css instead of `box-shadow`.

**`example`**
```
dropShadow(theme.shadows.shadow_04)
 => 'drop-shadow(0px 2px 1px rgba(0, 0, 0, 0.2)) 
         drop-shadow(0px 1px 1px rgba(0, 0, 0, 0.14)) 
         drop-shadow(0px 1px 3px rgba(0, 0, 0, 0.12))'
```

#### Parameters

| Name | Type | Default value | Description |
| :------ | :------ | :------ | :------ |
| `shadow` | `string` | `undefined` | box-shadow |
| `defaultShadow` | `string` | `'none'` | default shadow if none given |

#### Returns

`string`

___

### getShape

▸ `Const` **getShape**(`shapeProp`, `shapeContext`): ``"rounded"`` \| ``"squared"``

Get the shape or shape context in priority

#### Parameters

| Name | Type | Description |
| :------ | :------ | :------ |
| `shapeProp` | `ExtendedShapes` | shape value from props |
| `shapeContext` | ``"rounded"`` \| ``"squared"`` | shape value from context |

#### Returns

``"rounded"`` \| ``"squared"``

___

### splitShadows

▸ `Const` **splitShadows**(`shadows`, `defaultShadow`): `string`[]

Split the shadows in a given string

**`example`**
```
splitShadows('0px 2px 1px rgba(0, 0, 0, 0.2), 0px 1px 1px rgba(0, 0, 0, 0.14), 0px 1px 3px rgba(0, 0, 0, 0.12)')
 => [
     '0px 2px 1px rgba(0, 0, 0, 0.2)',
     '0px 1px 1px rgba(0, 0, 0, 0.14)',
     '0px 1px 3px rgba(0, 0, 0, 0.12)'
 ]
```

#### Parameters

| Name | Type | Description |
| :------ | :------ | :------ |
| `shadows` | `string` | shadow string from backyard |
| `defaultShadow` | `string` | default shadow if none given |

#### Returns

`string`[]

___

### useBackyardTheme

▸ `Const` **useBackyardTheme**(`props?`): [`BackyardTheme`](interfaces/BackyardTheme.md)

#### Parameters

| Name | Type |
| :------ | :------ |
| `props` | `UseBackyardThemeProps` |

#### Returns

[`BackyardTheme`](interfaces/BackyardTheme.md)

___

### validateBackyardTheme

▸ `Const` **validateBackyardTheme**(`theme`, `name`): `boolean`

#### Parameters

| Name | Type |
| :------ | :------ |
| `theme` | [`BackyardTheme`](interfaces/BackyardTheme.md) |
| `name` | `string` |

#### Returns

`boolean`
