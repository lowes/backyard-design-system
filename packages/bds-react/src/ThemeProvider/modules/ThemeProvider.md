# Namespace: ThemeProvider

## Variables

### bdsName

â€¢ **bdsName**: `string`
