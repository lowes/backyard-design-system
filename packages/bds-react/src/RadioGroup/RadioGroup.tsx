import * as React from 'react'

import FormGroup, { FormGroupProps, FormGroupRef } from '../FormGroup'
import useForkRef from '../utils/hooks/useForkRef'
import useControlled from '../utils/hooks/useControlled'
import type { BDSForwardRef } from '../utils/typings/BDSComponentProps'
import { useBackyardTheme, validateBackyardTheme } from '../ThemeProvider'

import RadioGroupContext from './RadioGroupContext'
import type { RadioGroupValue } from './RadioGroupContext'

/**
 * Backyard React Radio Group
 *
 * `RadioGroup` allows for grouping radio inputs and syncing them together.
 * It extends from `FormGroup` to provide extra functionality for `Radio` components.
 *
 * For `Checkbox` or `Switch` it is recommended to use `FormGroup`.
 *
 * This component is useful when combined with `FormControl` to
 * organize the visual structure of the form input. Example:
 *
 * ```
 *  <FormControl>
 *      <FormHeading>Heading</FormHeading>
 *      <RadioGroup
 *          name="radios"
 *          defaultValue="2"
 *          direction="row"
 *      >
 *          <FormControlLabel
 *              control={<Radio value="1" />}
 *              label="Label 1"
 *          />
 *          <FormControlLabel
 *              control={<Radio value="2" />}
 *              label="Label 2"
 *          />
 *          <FormControlLabel
 *              control={<Radio value="3" />}
 *              label="Label 3"
 *          />
 *      </FormGroup>
 *      <FormHelperText>Helper Text</FormHelperText>
 *  </FormControl>
 * ```
 */
const RadioGroup: BDSForwardRef<RadioGroupProps> = React.forwardRef<RadioGroupRef, RadioGroupProps>(
    function RadioGroup(
        { children, defaultValue, name, value: valueProp, onChange, ...props },
        ref,
    ) {
        // Get Backyard Theme Context
        const theme = useBackyardTheme()

        validateBackyardTheme(theme, 'RadioGroup')

        // Reference <FormControl>
        const rootRef = React.useRef(null)

        // Use Controlled to manage Uncontrolled behavior of the state of value
        // Determined through default value
        const [value, setValue] = useControlled({
            controlled: valueProp,
            default: defaultValue,
            name: `Backyard->RadioGroup->[name="${name}"]`,
        })

        // Fork forwarded ref and <FormControl> ref
        const handleRef = useForkRef(ref, rootRef)

        /**
         * Handles `onChange` event of `input`
         * Triggers defined `onChange` from props if available
         *
         * @param {Event} event - DOM event
         */
        const handleChange = (event) => {
            // Set value from target
            setValue(event.target.value)

            // If `onChange` prop defined,
            if (onChange) {
                // Trigger `onChange` function
                // Provides `Event` type and current target value
                onChange(event, event.target.value)
            }
        }

        const radioGroupValue: RadioGroupValue = {
            name,
            onChange: handleChange,
            value,
            defaultValue,
        }

        /**
         * Layout:
         *
         *  <RadioGroupContext.Provider>
         *      <FormGroup>
         *          ...
         *      </FormGroup>
         *  </RadioGroupContext.Provider>
         */
        return (
            <RadioGroupContext.Provider value={radioGroupValue}>
                <FormGroup role="radiogroup" ref={handleRef} {...props}>
                    {children}
                </FormGroup>
            </RadioGroupContext.Provider>
        )
    },
)

type RadioGroupRef = FormGroupRef

type RadioGroupOverrideProps = 'onChange'

interface RadioGroupProps extends Omit<FormGroupProps, RadioGroupOverrideProps> {
    /**
     * Children of `RadioGroup`, which have access to `RadioGroupContext`
     * Must contain at least one `Radio` child
     * Does not require `Radio` to be direct children
     */
    children?: React.ReactNode
    /**
     * Default Value of `RadioGroup`
     */
    defaultValue?: string
    /**
     * Required
     * DOM field `name` for all `Radio`s in child tree of `RadioGroup`
     */
    name: string
    /**
     * `onChange` function to trigger when `value` of `RadioGroupContext` changes
     * @argument {Event} event - DOM Event of `onChange`
     * @argument {boolean} value -  Value from DOM Event target
     */
    onChange?: (event: React.ChangeEvent, value: string) => void
    /**
     * Value of `RadioGroupContext`
     * Set by current active `Radio` child of `RadioGroup`
     */
    value?: string
}

RadioGroup.bdsName = 'RadioGroup'

export { RadioGroup }

export type { RadioGroupProps, RadioGroupRef }

export default RadioGroup
