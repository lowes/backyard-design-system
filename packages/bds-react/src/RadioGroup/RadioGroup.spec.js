import * as React from 'react'
import { render } from '../../test-utils'
import { ThemeProvider } from '../ThemeProvider'
import RadioGroup from './RadioGroup'

describe('RadioGroup Snapshots', () => {
    test('RadioGroup default snapshot', () => {
        const { asFragment } = render(
            <RadioGroup name="test-radio-group" />
        )

        expect(asFragment()).toMatchSnapshot()
    })

    describe('themes', () => {
        const themes = ['light', 'dark']

        themes.forEach((theme) => {
            test(`RadioGroup ${theme} theme snapshot`, () => {
                const { asFragment } = render(
                    <ThemeProvider theme={theme}>
                        <RadioGroup name="test-radio-group" />
                    </ThemeProvider>
                )

                expect(asFragment()).toMatchSnapshot()
            })
        })
    })

    describe('prop: disabled', () => {
        const disableds = [false, true]

        disableds.forEach((disabled) => {
            test(`RadioGroup ${disabled ? 'disabled' : 'enabled'} snapshot`, () => {
                const { asFragment } = render(
                    <RadioGroup name="test-radio-group" disabled={disabled} />
                )

                expect(asFragment()).toMatchSnapshot()
            })
        })
    })

    describe('extra props', () => {
        test('RadioGroup className snapshot', () => {
            const { asFragment } = render(
                <RadioGroup name="test-radio-group" className="test-class" />
            )

            expect(asFragment()).toMatchSnapshot()
        })

        test('RadioGroup DOM id, name, value snapshot', () => {
            const { asFragment } = render(
                <RadioGroup
                    id="test-id"
                    name="test-name"
                    value="test-value"
                />
            )

            expect(asFragment()).toMatchSnapshot()
        })
    })
})
