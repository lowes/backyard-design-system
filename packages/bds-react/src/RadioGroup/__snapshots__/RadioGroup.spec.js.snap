// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup Snapshots RadioGroup default snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0.direction--column {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0.direction--column > * {
  margin: 0.25rem 0;
}

.c0.direction--row {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-left: -0.25rem;
}

.c0.direction--row > * {
  margin: 0.25rem;
}

<div
    class="c0 form-group direction--column"
    role="radiogroup"
    spacing="size_4"
  />
</DocumentFragment>
`;

exports[`RadioGroup Snapshots extra props RadioGroup DOM id, name, value snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0.direction--column {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0.direction--column > * {
  margin: 0.25rem 0;
}

.c0.direction--row {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-left: -0.25rem;
}

.c0.direction--row > * {
  margin: 0.25rem;
}

<div
    class="c0 form-group direction--column"
    id="test-id"
    role="radiogroup"
    spacing="size_4"
  />
</DocumentFragment>
`;

exports[`RadioGroup Snapshots extra props RadioGroup className snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0.direction--column {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0.direction--column > * {
  margin: 0.25rem 0;
}

.c0.direction--row {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-left: -0.25rem;
}

.c0.direction--row > * {
  margin: 0.25rem;
}

<div
    class="c0 form-group direction--column test-class"
    role="radiogroup"
    spacing="size_4"
  />
</DocumentFragment>
`;

exports[`RadioGroup Snapshots prop: disabled RadioGroup disabled snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0.direction--column {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0.direction--column > * {
  margin: 0.25rem 0;
}

.c0.direction--row {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-left: -0.25rem;
}

.c0.direction--row > * {
  margin: 0.25rem;
}

<div
    class="c0 form-group direction--column"
    disabled=""
    role="radiogroup"
    spacing="size_4"
  />
</DocumentFragment>
`;

exports[`RadioGroup Snapshots prop: disabled RadioGroup enabled snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0.direction--column {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0.direction--column > * {
  margin: 0.25rem 0;
}

.c0.direction--row {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-left: -0.25rem;
}

.c0.direction--row > * {
  margin: 0.25rem;
}

<div
    class="c0 form-group direction--column"
    role="radiogroup"
    spacing="size_4"
  />
</DocumentFragment>
`;

exports[`RadioGroup Snapshots themes RadioGroup dark theme snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0.direction--column {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0.direction--column > * {
  margin: 0.25rem 0;
}

.c0.direction--row {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-left: -0.25rem;
}

.c0.direction--row > * {
  margin: 0.25rem;
}

<div
    class="c0 form-group direction--column"
    role="radiogroup"
    spacing="size_4"
  />
</DocumentFragment>
`;

exports[`RadioGroup Snapshots themes RadioGroup light theme snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0.direction--column {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0.direction--column > * {
  margin: 0.25rem 0;
}

.c0.direction--row {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-left: -0.25rem;
}

.c0.direction--row > * {
  margin: 0.25rem;
}

<div
    class="c0 form-group direction--column"
    role="radiogroup"
    spacing="size_4"
  />
</DocumentFragment>
`;
